### DISCUSSION TITLE
#39-DEV-product -> #35-DEV-branch 코드리뷰

### DISCUSSION THREAD

**Discussion by dev-wonny** [dev-wonny/coopang#128] [General]
[DEV-product -> DEV-branch PR](https://github.com/dev-wonny/coopang/pull/127)

---

# 크리티컬 이슈

##  product-branch 자체에서 컴파일 오류나는중
- 이게 제일 문제임
![image](https://github.com/user-attachments/assets/d6c45c35-0b23-4bf7-901e-f4b2dfe330ba)

## 상품 조회시 왜 재고 안가져오나요
![image](https://github.com/user-attachments/assets/d75bf8c8-4df6-4305-a84f-54263ae52609)

# 이슈
## 권한 체크는 컨트롤러에서
![image](https://github.com/user-attachments/assets/a198fd0f-3a10-4bcf-9f98-4e13ae09e4af)

## 전체적으로, 정렬 안됨
![image](https://github.com/user-attachments/assets/cb70489b-a39b-4fc9-8970-e92018b76ffe)

## @Data 있는데  getIsHidden getIsSale??
![image](https://github.com/user-attachments/assets/6a3c0deb-75fe-49c5-bf35-e3e6f6b7f02a)

## 접근제한자 없음
![image](https://github.com/user-attachments/assets/e4b69553-097e-48a4-8dc4-b967bbc493e8)

## ProductStockEntity, ProductStock 왜 있나요?
![image](https://github.com/user-attachments/assets/b063a1ef-f8c3-439f-b9ee-f8e6c106f619)
![image](https://github.com/user-attachments/assets/703be25f-2912-4355-bd58-0e77cd2deacd)

## 패키지이름 대문자 제거
package com.coopang.product.domain.entity.productStock;

## ProductRepositoryCustomImpl
![image](https://github.com/user-attachments/assets/f48fba9c-80ca-4eea-94d4-f332450c1d83)
- hubEntity.isDeleted.eq(condition.getIsDeleted()) 이런 조건 넣어야함
    
## ProductStockJpaRepository -> CustomImpl 없음
![image](https://github.com/user-attachments/assets/03ff60ed-5e39-49a7-8aa9-50c9fba4b6b1)

**Comment by bongdaehyun** [comment]
# 코드리뷰 답변

1. ai-communication의 값을 가져오면 해결됨( 커밋전략에 대해서 제대로 이해못해서 merge를 하지 않았습니다)
2. 상품조회시 안에 가져오는 코드를 보시면 join으로 가져와서 재고까지 같이 보입니다.
<img width="590" alt="image" src="https://github.com/user-attachments/assets/302281da-92b7-4766-b4d7-2b143eceee68">

3,4. 리팩토링할시에 고치도록할게용
5. boolean값은 is가 접두사에 있으면 getter가 작동하지않습니다. https://velog.io/@pkjeogus/08.30-TIL-Lombok%EC%9D%98-getter
6. record는 dto를 간편하게 만들기위함이라 접근제한자 설정이 없습니다.
gpt
<img width="318" alt="image" src="https://github.com/user-attachments/assets/73aa0141-4638-4a57-b5f7-1b7ec7794c2a">
7. 재고는 value Object라고 생각하여 따로 작성했습니당
<img width="319" alt="image" src="https://github.com/user-attachments/assets/c7c32b12-4e0d-4d71-a70a-cc94dcc91fa9">

8. 재고 기록이나 재고는 엔티티설정에 보시면 is deleted 만보이도록 설정해놨습니당
9. stockEntity관련해서는 search가 필요없어서 안만들었습니다.
